### PR TITLE
Implement Build and Flash commands

### PR DESCRIPTION
**Pull request by kyleparrott**
This extends beyond PR #2  so it includes some of the same changes that haven't been merged yet.

### CODE REVIEW

File: kubos-sdk.py [hunk 1]
```diff
@@ -60,6 +64,31 @@ def _init(name):
     c.description['homepage'] = 'https://<homepage>'
     init.initNonInteractive(None, c)
 
+def _build(verbose):
+    globalconf.set('plain', False)
```
Comment (marshall) [inline kubos-sdk.py]: I see this being set in a few different places, is the purpose to force to give pretty color printed output on the console?

File: kubos-sdk.py [hunk 2]
```diff
@@ -60,6 +64,31 @@ def _init(name):
...
+                                  cmake_generator='Ninja',
+                                  debug=None,
+                                  generate_only=False,
+                                  interactive=False,
+                                  target=current_target,
+                                  plain=False,
+                                  release_build=True,
+                                  registry=None)
+  
+    if verbose:
```
Comment (marshall) [inline kubos-sdk.py]: do we need to provide our own `--verbose` flag when passing through `unknown_args` would work just as well?

File: kubos.py [hunk 3]
```diff
@@ -3,93 +3,150 @@
...
-	elif command == 'update':
-		update()
+    parser = argparse.ArgumentParser('Kubos SDK')
+    subparser = parser.add_subparsers(dest='command', help='Available Kubos-sdk commands')
+
+    build_parser  = subparser.add_parser('build', help='build the project in the current directory')
+    flash_parser  = subparser.add_parser('flash', help='launch the built executable')
+    init_parser   = subparser.add_parser('init', help='initialize a new kubos project in the current directory')
+    target_parser = subparser.add_parser('target', help='set or display the current target hardware platform')
+    update_parser = subparser.add_parser('update', help='pull latest kubos-sdk docker container')
```
Comment (marshall) [inline kubos.py]: look at you, column aligning 👯

File: kubos.py [hunk 4]
```diff
@@ -3,93 +3,150 @@
...
+        return Client(base_url='unix://var/run/docker.sock') #use with docker-engine
+    elif sys.platform.startswith('darwin'):
+        return docker.from_env(assert_hostname=False) #use with docker-machine
+
+def get_project_name():
+    module_file_path = os.path.join(os.getcwd(), module_file_name)
+    if os.path.isfile(module_file_path):
+         with open(module_file_path, 'r') as module_file:
+            data = json.load(module_file)
+            name = str(data['name'])
```
Comment (marshall) [inline kubos.py]: is `data['name']` not already a `str` here?

File: kubos.py [hunk 5]
```diff
@@ -3,93 +3,150 @@
...
+
+        if not project_name:
+            print >>sys.stderr, 'Error: No module.json file found. Run "kubos build" in the root directory of your project'
+            sys.exit(1)
+
+        install_dir = os.path.dirname(os.path.realpath(__file__))
+        exe_path =  os.path.join(os.getcwd(), 'build', current_target, 'source', project_name)
+
+        if current_target.startswith('stm32'):
+            flash_script_path = str(os.path.join(install_dir, 'flash', 'openocd', 'flash.sh'))
```
Comment (marshall) [inline kubos.py]: this call to `str`, and below on line 119 are unnecessary

Comment (marshall) [inline kubos.py]: this call to `str`, and below on line 119 are unnecessary

**Comment by marshall** [comment]
@kyleparrott done with this review pass.. I only looked at the last 2 commits in the PR

**Comment by kyleparrott** [comment]
@marshall, should be ready for another pass.

File: flash/mspdebug/flash.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
```
Comment (marshall) [inline flash/mspdebug/flash.sh]: nit: add Apache 2.0 license headers to these shell scripts and it looks like we missed it on the osxusb.py script as well, go ahead and add it there :)

**Comment by marshall** [comment]
Looks good to merge. I realized we're also missing 'clean' and 'config' yotta commands, but I'd be ok to punt that to next release

**Comment by kyleparrott** [comment]
Ok, I am merging now and will look into clean and config commands for the next release.

**Pull request merged**